diff --git a/Source/WebCore/Modules/WebGPU/Implementation/Block.h b/Source/WebCore/Modules/WebGPU/Implementation/Block.h
new file mode 100644
index 00000000000..e9bbd01b572
--- /dev/null
+++ b/Source/WebCore/Modules/WebGPU/Implementation/Block.h
@@ -0,0 +1,23 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+inline void* _NSConcreteMallocBlock[32] = { };
+
+static inline void* _Block_copy(const void* block)
+{
+    return const_cast<void*>(block);
+}
+
+static inline void _Block_release(const void*)
+{
+}
+
+#define Block_copy(block) _Block_copy(block)
+#define Block_release(block) _Block_release(block)
+
+#ifdef __cplusplus
+}
+#endif
diff --git a/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp b/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp
index ddf71eb0d5b..614ef55e855 100644
--- a/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp
+++ b/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp
@@ -103,7 +103,11 @@ void DeviceImpl::destroy()
 
 RefPtr<XRBinding> DeviceImpl::createXRBinding()
 {
+#if PLATFORM(WIN)
+    return nullptr;
+#else
     return XRBindingImpl::create(adoptWebGPU(wgpuDeviceCreateXRBinding(m_backing.get())), m_convertToBackingContext);
+#endif
 }
 
 RefPtr<Buffer> DeviceImpl::createBuffer(const BufferDescriptor& descriptor)
@@ -186,10 +190,12 @@ static WGPUColorSpace NODELETE convertToWGPUColorSpace(const PredefinedColorSpac
     return WGPUColorSpace::SRGB;
 }
 
+#if PLATFORM(COCOA) && ENABLE(VIDEO)
 void DeviceImpl::updateExternalTexture(const WebCore::WebGPU::ExternalTexture&, const WebCore::MediaPlayerIdentifier&)
 {
     RELEASE_ASSERT_NOT_REACHED();
 }
+#endif
 
 RefPtr<ExternalTexture> DeviceImpl::importExternalTexture(const ExternalTextureDescriptor& descriptor)
 {
diff --git a/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.h b/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.h
index 0d500e117df..344d5519585 100644
--- a/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.h
+++ b/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.h
@@ -73,7 +73,9 @@ private:
     RefPtr<Texture> createTexture(const TextureDescriptor&) final;
     RefPtr<Sampler> createSampler(const SamplerDescriptor&) final;
     RefPtr<ExternalTexture> importExternalTexture(const ExternalTextureDescriptor&) final;
+#if PLATFORM(COCOA) && ENABLE(VIDEO)
     void NODELETE updateExternalTexture(const WebCore::WebGPU::ExternalTexture&, const WebCore::MediaPlayerIdentifier&) final;
+#endif
 
     RefPtr<BindGroupLayout> createBindGroupLayout(const BindGroupLayoutDescriptor&) final;
     RefPtr<PipelineLayout> createPipelineLayout(const PipelineLayoutDescriptor&) final;
diff --git a/Source/WebCore/Sources.txt b/Source/WebCore/Sources.txt
index 544e39fd3ce..d830e0379cf 100644
--- a/Source/WebCore/Sources.txt
+++ b/Source/WebCore/Sources.txt
@@ -84,10 +84,6 @@ Modules/WebGPU/Implementation/WebGPUSamplerImpl.cpp
 Modules/WebGPU/Implementation/WebGPUShaderModuleImpl.cpp
 Modules/WebGPU/Implementation/WebGPUTextureImpl.cpp
 Modules/WebGPU/Implementation/WebGPUTextureViewImpl.cpp
-Modules/WebGPU/Implementation/WebGPUXRBindingImpl.cpp
-Modules/WebGPU/Implementation/WebGPUXRProjectionLayerImpl.cpp
-Modules/WebGPU/Implementation/WebGPUXRSubImageImpl.cpp
-Modules/WebGPU/Implementation/WebGPUXRViewImpl.cpp
 Modules/WebGPU/WGSLLanguageFeatures.cpp
 Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.cpp
 Modules/applicationmanifest/ApplicationManifestParser.cpp
