diff --git a/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp b/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp
index e75f6a7..d244c0a 100644
--- a/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp
+++ b/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp
@@ -30,13 +30,15 @@
 
 #include "GPUConnectionToWebProcess.h"
 #include "Logging.h"
+#if ENABLE(GPU_PROCESS_MODEL)
 #include "MeshImpl.h"
 #include "ModelObjectHeap.h"
+#include "RemoteMesh.h"
+#endif
 #include "RemoteAdapter.h"
 #include "RemoteCompositorIntegration.h"
 #include "RemoteGPUMessages.h"
 #include "RemoteGPUProxyMessages.h"
-#include "RemoteMesh.h"
 #include "RemotePresentationContext.h"
 #include "RemoteRenderingBackend.h"
 #include "StreamServerConnection.h"
@@ -72,7 +74,9 @@ RemoteGPU::RemoteGPU(WebGPUIdentifier identifier, GPUConnectionToWebProcess& gpu
     , m_workQueue(IPC::StreamConnectionWorkQueue::create("WebGPU work queue"_s))
     , m_streamConnection(WTF::move(streamConnection))
     , m_objectHeap(WebGPU::ObjectHeap::create())
+#if ENABLE(GPU_PROCESS_MODEL)
     , m_modelObjectHeap(ModelObjectHeap::create())
+#endif
     , m_identifier(identifier)
     , m_renderingBackend(renderingBackend)
 {
@@ -135,7 +139,9 @@ void RemoteGPU::workQueueUninitialize()
     streamConnection->invalidate();
     m_streamConnection = nullptr;
     protect(m_objectHeap)->clear();
+#if ENABLE(GPU_PROCESS_MODEL)
     protect(m_modelObjectHeap)->clear();
+#endif
     m_backing = nullptr;
 }
 
@@ -313,7 +319,7 @@ static RefPtr<WebKit::Mesh> createModelBackingInternal(unsigned width, unsigned
 }
 #endif
 
+#if ENABLE(GPU_PROCESS_MODEL)
 void RemoteGPU::createModelBacking(unsigned width, unsigned height, const WebModel::ImageAsset& diffuseTexture, const WebModel::ImageAsset& specularTexture, WebModelIdentifier identifier, CompletionHandler<void(Vector<MachSendRight>&&)>&& callback)
 {
-#if ENABLE(GPU_PROCESS_MODEL)
     assertIsCurrent(workQueue());
@@ -326,15 +332,8 @@ void RemoteGPU::createModelBacking(unsigned width, unsigned height, const WebMod
     auto mesh = createModelBackingInternal(width, height, diffuseTexture, specularTexture, gpuProcessConnection->webProcessIdentity(), WTF::move(callback));
     auto remoteMesh = RemoteMesh::create(*m_gpuConnectionToWebProcess.get(), *this, *mesh, objectHeap, protect(*m_streamConnection), identifier);
     objectHeap->addObject(identifier, remoteMesh);
-#else
-    UNUSED_PARAM(width);
-    UNUSED_PARAM(height);
-    UNUSED_PARAM(diffuseTexture);
-    UNUSED_PARAM(specularTexture);
-    UNUSED_PARAM(identifier);
-    UNUSED_PARAM(callback);
-#endif
 }
+#endif
 
 void RemoteGPU::isValid(WebGPUIdentifier identifier, CompletionHandler<void(bool, bool)>&& completionHandler)
 {
diff --git a/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.h b/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.h
index 3943883..aa136ba 100644
--- a/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.h
+++ b/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.h
@@ -27,7 +27,9 @@
 
 #if ENABLE(GPU_PROCESS)
 
+#if ENABLE(GPU_PROCESS_MODEL)
 #include "ModelObjectHeap.h"
+#endif
 #include "RemoteGPURequestAdapterResponse.h"
 #include "RemoteVideoFrameIdentifier.h"
 #include "SharedPreferencesForWebProcess.h"
@@ -129,7 +131,9 @@ private:
 
 
     void requestAdapter(const WebGPU::RequestAdapterOptions&, WebGPUIdentifier, CompletionHandler<void(std::optional<RemoteGPURequestAdapterResponse>&&)>&&);
+#if ENABLE(GPU_PROCESS_MODEL)
     void NODELETE createModelBacking(unsigned width, unsigned height, const WebModel::ImageAsset& diffuseTexture, const WebModel::ImageAsset& specularTexture, WebKit::WebModelIdentifier, CompletionHandler<void(Vector<MachSendRight>&&)>&&);
+#endif
 
     void createPresentationContext(const WebGPU::PresentationContextDescriptor&, WebGPUIdentifier);
 
@@ -143,7 +147,9 @@ private:
     RefPtr<IPC::StreamServerConnection> m_streamConnection;
     RefPtr<WebCore::WebGPU::GPU> m_backing WTF_GUARDED_BY_CAPABILITY(workQueue());
     Ref<WebGPU::ObjectHeap> m_objectHeap WTF_GUARDED_BY_CAPABILITY(workQueue());
+#if ENABLE(GPU_PROCESS_MODEL)
     Ref<ModelObjectHeap> m_modelObjectHeap WTF_GUARDED_BY_CAPABILITY(workQueue());
+#endif
     const WebGPUIdentifier m_identifier;
     Ref<RemoteRenderingBackend> m_renderingBackend;
 };
