--- a/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp
+++ b/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp
@@ -32,0 +33 @@
+#if ENABLE(GPU_PROCESS_MODEL)
@@ -34,0 +36,2 @@
+#include "RemoteMesh.h"
+#endif
@@ -39 +41,0 @@
-#include "RemoteMesh.h"
@@ -74,0 +77 @@
+#if ENABLE(GPU_PROCESS_MODEL)
@@ -75,0 +79 @@
+#endif
@@ -137,0 +142 @@
+#if ENABLE(GPU_PROCESS_MODEL)
@@ -138,0 +144 @@
+#endif
@@ -315,0 +322 @@
+#if ENABLE(GPU_PROCESS_MODEL)
@@ -318 +324,0 @@
-#if ENABLE(GPU_PROCESS_MODEL)
@@ -329,8 +334,0 @@
-#else
-    UNUSED_PARAM(width);
-    UNUSED_PARAM(height);
-    UNUSED_PARAM(diffuseTexture);
-    UNUSED_PARAM(specularTexture);
-    UNUSED_PARAM(identifier);
-    UNUSED_PARAM(callback);
-#endif
@@ -337,0 +336 @@
+#endif
--- a/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.h
+++ b/Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.h
@@ -29,0 +30 @@
+#if ENABLE(GPU_PROCESS_MODEL)
@@ -30,0 +32 @@
+#endif
@@ -131,0 +134 @@
+#if ENABLE(GPU_PROCESS_MODEL)
@@ -132,0 +136 @@
+#endif
@@ -145,0 +150 @@
+#if ENABLE(GPU_PROCESS_MODEL)
@@ -146,0 +152 @@
+#endif
