diff --git a/Source/WTF/wtf/PlatformEnable.h b/Source/WTF/wtf/PlatformEnable.h
--- a/Source/WTF/wtf/PlatformEnable.h
+++ b/Source/WTF/wtf/PlatformEnable.h
@@ -629,6 +629,10 @@
 #define ENABLE_WEBGPU PLATFORM(COCOA)
 #endif
 
+#if !defined(ENABLE_WEBGPU_BY_DEFAULT) && PLATFORM(WIN) && ENABLE_WEBGPU
+#define ENABLE_WEBGPU_BY_DEFAULT 1
+#endif
+
 #if !defined(ENABLE_WEBGPU_BY_DEFAULT) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 260000) || (PLATFORM(IOS)) || (PLATFORM(VISION)))
 #define ENABLE_WEBGPU_BY_DEFAULT 1
 #endif
