diff --git a/Source/WebCore/Modules/WebGPU/Implementation/WebGPUImpl.cpp b/Source/WebCore/Modules/WebGPU/Implementation/WebGPUImpl.cpp
index 2420ff0..d2f2d06 100644
--- a/Source/WebCore/Modules/WebGPU/Implementation/WebGPUImpl.cpp
+++ b/Source/WebCore/Modules/WebGPU/Implementation/WebGPUImpl.cpp
@@ -96,7 +96,7 @@ void GPUImpl::requestAdapter(const RequestAdapterOptions& options, CompletionHan
         .nextInChain = nullptr,
         .featureLevel = WGPUFeatureLevel_Undefined,
         .powerPreference = options.powerPreference ? convertToBackingContext->convertToBacking(*options.powerPreference) : static_cast<WGPUPowerPreference>(WGPUPowerPreference_Undefined),
-        .forceFallbackAdapter = options.forceFallbackAdapter,
+        .forceFallbackAdapter = true,
         .backendType = WGPUBackendType_D3D12,
         .compatibleSurface = nullptr,
     };
